test/unit/specs/utils/dom/awaitSelector.spec.js

```diff
@@ -31,39 +31,40 @@ describe("DOM::awaitSelector", () => {
     removeNode(nodes[0]);
   };
 
-  const awaitSelectorAndAssert = (id, win, doc, done) => {
+  const awaitSelectorAndAssert = (id, win, doc) => {
     const result = awaitSelector(`#${id}`, selectNodes, 1000, win, doc);
 
     createAndAppendNodeDelayed(id);
 
-    result
+    return result
       .then(nodes => {
-        done();
-        cleanUp(id);
         expect(nodes[0].tagName).toEqual("DIV");
       })
-      .catch(() => {
-        done();
+      .finally(() => {
         cleanUp(id);
-        fail(`${id} should be found`);
```

Comment:
This one had a couple issues. First, we were calling `done()` here and then `fail`, which didn't really make sense, and the fact that we were calling `done` before cleaning up meant we were potentially attempting to clean up DOM elements after Jasmine moved onto the next test.